Source/options.cpp

```diff
@@ -712,13 +652,152 @@ std::vector<OptionEntryBase *> ChatOptions::GetEntries()
+		language = SCE_SYSTEM_PARAM_LANG_ENGLISH_US; // default to english
+	std::string locale = std::string(vita_locales[language]);
+	sceAppUtilShutdown();
+#elif defined(__3DS__)
+	std::string locale = n3ds::GetLocale();
+#else
+	std::string locale = std::locale("").name().substr(0, 5);
+#endif
+
+	locale = locale.substr(0, 5);
+	LogVerbose("Prefered locale: {}", locale);
+	if (!HasTranslation(locale)) {
+		locale = locale.substr(0, 2);
+		if (!HasTranslation(locale)) {
+			locale = "en";
+		}
+	}
+	LogVerbose("Best match locale: {}", locale);
+
+	GetIniValue(category.data(), key.data(), szCode, sizeof(szCode), locale.c_str());
+}
+void OptionEntryLanguageCode::SaveToIni(string_view category) const
+{
+	SetIniValue(category.data(), key.data(), szCode);
+}
+
+void OptionEntryLanguageCode::CheckLanguagesAreInitialized() const
+{
+	if (languages.size() != 0)
+		return;
+
+	// Add well-known supported languages
+	languages.push_back({ "bg", "Bulgarian" });
+	languages.push_back({ "cs", "Czech" });
+	languages.push_back({ "da", "Danish" });
+	languages.push_back({ "de", "German" });
+	languages.push_back({ "en", "English" });
+	languages.push_back({ "es", "Spanish" });
+	languages.push_back({ "fr", "French" });
+	languages.push_back({ "ja", "Japanese" });
+	languages.push_back({ "hr", "Croatian" });
+	languages.push_back({ "it", "Italian" });
+	languages.push_back({ "ko_KR", "Korean" });
+	languages.push_back({ "pl", "Polish" });
+	languages.push_back({ "pt_BR", "Portuguese" });
+	languages.push_back({ "ro_RO", "Romansh" });
+	languages.push_back({ "ru", "Russian" });
+	languages.push_back({ "sv", "Swedish" });
+	languages.push_back({ "uk", "Ukrainian" });
+	languages.push_back({ "zh_CN", "Simplified Chinese" });
+	languages.push_back({ "zh_TW", "Traditional Chinese" });
+
+	// Ensures that the ini specified language is present in languages list even if unkown (for example if someone starts to translate a new language)
+	if (std::find_if(languages.begin(), languages.end(), [this](const auto &x) { return x.first == this->szCode; }) == languages.end()) {
+		languages.push_back({ szCode, szCode });
+	}
+}
+
+size_t OptionEntryLanguageCode::GetListSize() const
+{
+	CheckLanguagesAreInitialized();
+	return languages.size();
+}
+string_view OptionEntryLanguageCode::GetListDescription(size_t index) const
+{
+	CheckLanguagesAreInitialized();
+	return languages[index].second;
+}
+size_t OptionEntryLanguageCode::GetActiveListIndex() const
+{
+	CheckLanguagesAreInitialized();
+	auto found = std::find_if(languages.begin(), languages.end(), [this](const auto &x) { return x.first == this->szCode; });
+	if (found == languages.end())
+		return 0;
+	return std::distance(languages.begin(), found);
+}
+void OptionEntryLanguageCode::SetActiveListIndex(size_t index)
+{
+	strcpy(szCode, languages[index].first.c_str());
+	NotifyValueChanged();
+}
+
+string_view OptionEntryLanguageCode::operator*() const
+{
+	return szCode;

```

Comment:
Please move this to the header so that the compiler can inline it even without LTO. We access this often via `IsSmallFontTall`.

Reply:
I will open a new pr for the `operator*()` in other OptionEntryBase based classes. 🙂